src/util/acl.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * @copyright Copyright (c) 2023 Daniel Kesselberg <mail@danielkesselberg.de>
+ *
+ * @author Daniel Kesselberg <mail@danielkesselberg.de>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/**
+ * @param {object} mailbox object
+ * @param {string} rights list of rights to check
+ * @return {boolean}
+ */
+export function mailboxHasRights(mailbox, rights) {
+	if (!mailbox.myAcls) {
+		return true
+	}
+
+	const acls = [...mailbox.myAcls]
+
+	for (const right of [...rights]) {
+		if (!acls.includes(right)) {
+			return false
+		}
+	}
+
+	return true
+}
```

Comment:
Epic! I wanted to apply something like that to the existing ACL code because there is a lot of code duplication.

Reply:
My thoughts exactly! I can do it after all the issues are closed, in one big pr, if you dont mind

Reply:
Thank you!